### ISSUE TITLE
Problem with `geom_ribbon` and alpha combined with `coord-cartesian` in R 4.1.0 on Windows

### ISSUE THREAD

**Issue by NikNakk** [tidyverse/ggplot2#4498]
Since upgrading to R 4.1.0, I've run into an issue with `ggplot2` on Windows where setting an `xlim` using `coord_cartesian` can result in a `geom_ribbon` being suppressed altogether. I've been able to reproduce this on two different Windows 10 machines running R 4.1.0, both inside RStudio and RGui. Running the code below without the `coord_cartesian` works fine on all machines I've tested. With the `coord_cartesian`, it produces a plot with no ribbon on my two Windows 10 machines using R 4.1.0, but works fine on R 4.0.5 on Windows 10 as well as R 4.1.0 tested using Ubuntu 20.04. Note removing the `alpha` parameter makes the issue go away even on R 4.1.0 on Windows 10.

I first ran into this using the `survminer` package, but I've managed to get a reprex now which occurs just in `ggplot2`.

``` r
library(ggplot2); ggplot(data.frame(x = 1:10, ymin = 1:10, ymax = 10 + 1:10), aes(x)) + geom_ribbon(aes(ymin = ymin, ymax = ymax), alpha = 0.5) + coord_cartesian(c(1, 9))
```

![](https://i.imgur.com/3si3jAB.png)

<sup>Created on 2021-06-01 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by yutannihilation** [comment]
Thanks for reporting, confirmed on my Windows. A quick workaround is to use AGG as the graphic backend of RStudio, but this sounds a bit serious in that this happens not only on RStudio, but also on RGui and even with `ggsave()`.

**Comment by clauswilke** [comment]
Isn't this just again the problem that the graphics device on Windows can't clip properly?

**Comment by yutannihilation** [comment]
Hmm..., I thought it was the problem only with RStudio's Plot pane, but my memory might be wrong (I haven't been used Windows for years), so let me check...

**Comment by clauswilke** [comment]
Like e.g. #4029.

**Comment by yutannihilation** [comment]
Ah, true. Sorry, I was a bit confused.

@NikNakk 
> works fine on R 4.0.5 on Windows 10

Sorry, this seems a known issue. My guess is this is because you used AGG as the backend on RStudio with R 4.0.5, and after migrating to R 4.1.0 you don't install it yet?

**Comment by NikNakk** [comment]
No, it works fine on vanilla R 4.0.5 and RStudio (i.e. using the default graphics device for RStudio). On R 4.1.0 it doesn't work using any graphics device I've tried. I've just reproduced it using RGui.exe and using the png device. Identical code run on R 4.0.5 and R 4.1.0 (with no RStudio running) gives the following.

**Comment by NikNakk** [comment]
I've also put together a very simple R package to reproduce this issue. Artefacts from Appveyor are available at https://ci.appveyor.com/project/NikNakk/ggplot2missingribbon/build/artifacts

Package itself is at https://github.com/NikNakk/ggplot2missingribbon

Note all it does is to generate a test PNG. On Appveyor, R 4.0.5 produces the correct output (and so 'fails' the check, because the snapshot is based on R 4.1.0), whereas R 4.1.0 produces an incorrect PNG, as does r-devel.

**Comment by yutannihilation** [comment]
Thanks for the detailed testing! I also confirmed this with my Windows with R 4.0.4 (sorry for my laziness...).

It seems this is a version of the known issue #4029, something related to alpha and clipping. If so, there's nothing we can do on ggplot2's side. Please use AGG as the graphic backend of RStudio.

I hope we can reproduce this with grid so that we can file a bug report to the R bugzilla.

**Comment by NikNakk** [comment]
Ok, fair enough, though it's odd that it reproducibly works fine for me on R 4.0.5 and not on R 4.1.0 across both my own machines and appveyor. For my personal use, I'll switch to ragg/AGG as suggested.

**Comment by yutannihilation** [comment]
Yeah, it's a bit worrisome that the situation seems to get worse in R 4.1.0...

**Comment by clauswilke** [comment]
Maybe we should copy @pmur002. Though we would probably need a reproducible example using only grid code to make further progress from here.

**Comment by FBartos** [comment]
Hi, encountered the same issue. Interestingly, it also happens when using ``ggplot2::alpha()`` to fill base plot polygons for example.

**Comment by bergsmat** [comment]
I have this problem too.

sessionInfo()
R Under development (unstable) (2021-06-07 r80459)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

**Comment by tamas-ferenci** [comment]
As a sidenote, for a workaround for shiny you'll need [this](https://github.com/r-lib/ragg/issues/74#issuecomment-779918044).

**Comment by JWiley** [comment]
I am getting this as well, although it only applies when `ylim` clips the ribbon, not when `xlim` clips it. This is on:

R version 4.1.0 (2021-05-18)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

using Emacs so the default graphics device. `pdf()` works fine.

**Comment by pmur002** [comment]
My timing is out - I have committed a fix (to semitransparent drawing on Windows) to r-devel (r81114) that I think fixes this.

**Comment by yutannihilation** [comment]
Wow, thanks so much for fixing! 🎉
